src/main/java/net/pms/dlna/DLNAMediaInfo.java

```diff
@@ -2438,48 +2443,74 @@ public void setReferenceFrameCount(byte referenceFrameCount) {
 	}
 
 	/**
-	 * @return AVC level for video stream or {@code null} if not parsed.
+	 * @return The AVC/H.264 level for video stream or {@code null} if not
+	 *         parsed/relevant.
 	 */
-	public String getAvcLevel() {
-		avcLevelLock.readLock().lock();
-		try {
-			return avcLevel;
-		} finally {
-			avcLevelLock.readLock().unlock();
-		}
+	public H264Level getH264Level() {
+		return H264Level.typeOf(getVideoFormatProfile());
 	}
 
 	/**
-	 * Sets AVC level for video stream or {@code null} if not parsed.
+	 * @return The HEVC/H.265 level for video stream or {@code null} if not
+	 *         parsed/relevant.
+	 */
+	public H265Level getH265Level() {
+		return H265Level.typeOf(getVideoFormatProfile());
+	}
+
+	/**
+	 * Gets the video format profile for the video stream or {@code null} if not parsed/relevant.
 	 *
-	 * @param avcLevel AVC level.
+	 * @return the video format profile {@link String}.
 	 */
-	public void setAvcLevel(String avcLevel) {
-		avcLevelLock.writeLock().lock();
+	public String getVideoFormatProfile() {
+		videoFormatProfileLock.readLock().lock();
 		try {
-			this.avcLevel = avcLevel;
+			return videoFormatProfile;
 		} finally {
-			avcLevelLock.writeLock().unlock();
+			videoFormatProfileLock.readLock().unlock();
 		}
 	}
 
-	public int getAvcAsInt() {
+	/**
+	 * Sets video format profile for the video stream.
+	 *
+	 * @param formatProfile AVC level.
+	 */
+	public void setVideoFormatProfile(String formatProfile) {
+		videoFormatProfileLock.writeLock().lock();
 		try {
-			return Integer.parseInt(getAvcLevel().replaceAll("\\.", ""));
-		} catch (Exception e) {
-			return 0;
+			this.videoFormatProfile = formatProfile == null ? null : formatProfile.trim();
+		} finally {
+			videoFormatProfileLock.writeLock().unlock();
 		}
 	}
 
+	/**
+	 * @return The AVC/H.264 profile for video stream or {@code null} if not
+	 *         parsed/relevant.
+	 */
 	public String getH264Profile() {
-		synchronized (h264ProfileLock) {
-			return h264Profile;
+		String profile = substringBefore(lowerCase(getVideoFormatProfile(), Locale.ROOT), "@l");
+		if (isNotBlank(profile)) {
+			return profile;
+		} else {
```

Comment:
Just a formatting note, there's no need for the `else` here, or on line 2511, since it will return beforehand